common-practices-tools/security/yubikey.md

```diff
@@ -0,0 +1,309 @@
+
+This is required of CivicActions "privileged users" such as System Administrators, and it is our intention that it be standard practice for all CivicActions employees and contractors.
+
+##### Installing the Yubico libpam module
+
+In order to connect your YubiKey to the screen locking software on your computer, you need to:
+
+###### Arch yubico-pam setup
+
+```bash
+pacaur -S yubico-pam
+```
+
+###### Fedora Yubico libpam module
+
+###### Ubuntu/Xubuntu Yubico libpam module
+
+```bash
+sudo apt-get install libpam-yubico
+```
+
+##### Set up PAM TFA on Linux
+
+PAM is the Pluggable Authentication Module used by GNU/Linux and Mac OS X to manage login authentication.
+
+See [Yubico GitHub](https://github.com/Yubico/yubico-pam/blob/b0e243835e61418bfa760e57c3d313b2e9452e87/doc/Authentication_Using_Challenge-Response.adoc) page for complete documentation.
+
+```bash
+ykpamcfg -2 -v
+```
+
+Ubuntu autoconfiguration during installation of `libpam-yubico` may already have placed a line like the following in either `/etc/pam.d/common-auth` or `/etc/pam.d/system-auth`. If not using Ubuntu (or the line is not there), edit `/etc/pam.d/system-auth` (will need to `sudo`) and add the following line at the top of the file:
+
+```bash
+auth      required  pam_yubico.so   mode=challenge-response
+```
+
+##### YubiKey removal lock
+
+For additional security, you may want to immediately lock the screen when the YubiKey is removed.
+
+This locks the laptop immediately when any YubiKey is removed. If you are not using xautolock as your "away detector", replace xautolock with a command to trigger your screen lock with the "away detector" that you do use. This is inspired by <https://vtluug.org/wiki/Yubikey#Automatic_Screen_Locking_.28i3lock.2C_slock.2C_etc..29>
+
+As your login user, create executable file `~/bin/ykgone`:
+
+```bash
+#!/bin/bash
+USER=$(stat -c "%U" "$0")
+if usb-devices | fgrep Vendor=1050; then
+  echo "YubiKey present"
+else
+  echo "YubiKey not present, locking"
+  export DISPLAY=":0"
+  export XAUTHORITY=/home/$USER/.Xauthority
+  su $USER -c "xautolock -locknow" &
+fi
+```
+
+Next, create (with sudo) a device notification file `/etc/udev/rules.d/90-yubikey.rules`:
+
+```bash
+ACTION=="remove", ATTRS{idVendor}=="1050", RUN+="/home/$USER/bin/ykgone"
+```
+
+#### Personalize your YubiKey on Linux
+
+This allows you to use your YubiKey with Google TFA (new fangled U2F), as well as LastPass (which uses the OTP application).
+
+```bash
+neoman
+# Enable OTP, U2F, CCID checkboxes if needed, follow instructions to add and remove key
+
+ykpersonalize -2 -ochal-resp -ochal-hmac -ohmac-lt64 -oserial-api-visible
+```
+
+### Mac OS X
+
+This is the Mac OS X specific documentation for YubiKey.
+
+_Please help make this page more useful by adding links you found useful (describe exactly how they are useful) and specific steps you used to install, configure, and test your YubiKey._
+
+#### Personalize your YubiKey on MacOS
```

Comment:
I don't believe personalization is required anymore (as with neo manager above)

Reply:
Ok, I'm removing it.